tests/api_test.py

```diff
@@ -4502,8 +4502,31 @@ def test_pretty_print_unitvar(self):
     self.assertIn('in (*,)', str(jaxpr))
     self.assertNotIn('in (a,)', str(jaxpr))
 
-  def test_dce_jaxpr_scan(self):
-    raise unittest.SkipTest()  # TODO(mattjj)
+class DCETest(jtu.JaxTestCase):
```

Comment:
The main changes in this PR are the tests. The bug slipping through previously showed us we need better tests.